cmd/podman/create.go

```diff
@@ -388,14 +378,62 @@ func parseCreateOpts(ctx context.Context, c *cli.Context, runtime *libpod.Runtim
 		return nil, errors.Errorf("--cpu-quota and --cpus cannot be set together")
 	}
 
-	utsMode := container.UTSMode(c.String("uts"))
-	if !cc.IsNS(string(utsMode)) && !utsMode.Valid() {
+	// Kernel Namespaces
+	var pod *libpod.Pod
+	if c.IsSet("pod") {
+		pod, err = runtime.LookupPod(c.String("pod"))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	pidModeStr := c.String("pid")
+	if !c.IsSet("pid") && pod != nil && pod.SharesPID() {
+		pidModeStr = "pod"
+	}
+	pidMode := container.PidMode(pidModeStr)
+	if !cc.IsNS(pidModeStr) && !cc.IsPod(pidModeStr) && !pidMode.Valid() {
```

Comment:
Can we modify Valid() for all of these so it considers IsPod and IsNS as valid?

Reply:
We currently get our valid from: vendor/github.com/docker/docker/api/types/container/host_config.go . I can reformat it so it's cc.Valid(pidMode) instead, and include the other checks as well as pidMode.Valid() ?

Reply:
SGTM